src/workflow.ts

```diff
@@ -60,11 +74,21 @@ export const StatusString = {
   ERROR: "ERROR",
 } as const;
 
+export type WorkflowOf<WC, T extends unknown[], R> = {
+  [K in keyof WC]: WC[K] extends Workflow<T, R> ? WC[K] : never
+}[keyof WC];
+
 export interface WorkflowContext extends DBOSContext {
+  invoke<T extends object>(targetCfg: ConfiguredClass<T>): InvokeFuncsConf<T>;
```

Comment:
If I understand correctly, the two `invoke`s are actually the same from the user perspective--the only difference is whether they pass in the class itself or an instantiation of the class. That's great. But can something similar be done for child workflows?

Reply:
And yes, I got talked into consolidating to one function at the end of last week.  Made sense.

Reply:
The documentation is something like:
Say you have:
`ctx.invoke(MyClass).function(arg); // Uh, what do I do with config info for MyClass?`
That becomes
```
const myClassConfig = initClassConfiguration(MyClass, 'whateverItNeeds');
ctx.invoke(myClassConfig).function(arg);  // OK got it
```

What complexity can be shaken loose from here?

Reply:
Like I said, for `invoke` the solution is good because the two are nearly the same from the user perspective. Is there some way to do something similar for child workflows?  For example, can you do `invokeChildworkflow(myClassConfig.function, args)` instead of `invokeChildWorkflow(myclassConfig, MyClass.function, args)`?

Reply:
> I think that for `invoke` the solution is good because the two are nearly the same from the user perspective. Is there some way to do something similar for child workflows? For example, can you do `invokeChildworkflow(myClassConfig.function, args)` instead of `invokeChildWorkflow(myclassConfig, MyClass.function, args)`?

No, because your syntax didn't provide the configuration.  `invokeChildWorkflow (old)` is the inconsistent one because the signature admits any function, class or not, and probably should have taken the class in the first place because every other invoke does.